src/overlays/actors/ovl_En_Anubice_Fire/z_en_anubice_fire.c

```diff
@@ -50,17 +53,213 @@ static ColliderCylinderInit D_809B31E0 = {
+            this->cylinder.base.atFlags |= 8;
+            this->cylinder.info.toucher.dmgFlags = 2;
+            this->unk_15A = 30;
+            this->actor.params = 1;
+            this->actor.velocity.x *= -1.0f;
+            this->actor.velocity.y *= -0.5f;
+            this->actor.velocity.z *= -1.0f;
+        } else {
+            this->unk_15A = 0;
+            EffectSsBomb2_SpawnLayered(globalCtx, &this->actor.world.pos, &sp78, &sp84, 10, 5);
+            this->actor.velocity.z = 0.0f;
+            this->actor.velocity.y = 0.0f;
+            this->actor.velocity.x = 0.0f;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_ANUBIS_FIREBOMB);
+            this->actionFunc = &func_809B2B48;
+        }
+    } else if (!(this->scale < .4f)) {
+        f32 scale = 1000.0f;
+        f32 life = 10.0f;
+
+        s32 i;
+        for (i = 0; i < 10; i++) {
+            pos.x = this->actor.world.pos.x + (Rand_ZeroOne() - 0.5f) * (this->scale * 20.0f);
+            pos.y = this->actor.world.pos.y + (Rand_ZeroOne() - 0.5f) * (this->scale * 20.0f);
+            pos.z = this->actor.world.pos.z;
+            EffectSsKiraKira_SpawnDispersed(globalCtx, &pos, &velocity, &accel, &primColor, &envColor, scale, life);
+        }
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_ANUBIS_FIRE - SFX_FLAG);
+    }
+}
+
+void func_809B2B48(EnAnubiceFire* this, GlobalContext* globalCtx) {
+    Vec3f velocity = { 0.0f, 0.0f, 0.0f };
+    Vec3f accel = { 0.0f, 0.0f, 0.0f };
+    Vec3f pos;
+    Color_RGBA8 primColor = { 255, 255, 0, 255 };
+    Color_RGBA8 envColor = { 255, 0, 0, 255 };
+    s32 pad;
+    s32 i;
+
+    if (this->unk_15C == 0) {
+        for (i = 0; i < 20; i++) {
+            pos.x = this->actor.world.pos.x;
+            pos.y = this->actor.world.pos.y;
+            pos.z = this->actor.world.pos.z;
+            accel.x = Rand_CenteredFloat(8.0f);
+            accel.y = Rand_CenteredFloat(2.0f);
+            accel.z = Rand_CenteredFloat(8.0f);
+            EffectSsKiraKira_SpawnDispersed(globalCtx, &pos, &velocity, &accel, &primColor, &envColor, 2000, 10);
+        }
+
+        this->unk_15C = 2;
+        this->unk_15E++;
+        if (this->unk_15E >= 6) {
+            Actor_Kill(&this->actor);
+        }
+    }
+}
+
+void EnAnubiceFire_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnAnubiceFire* this = THIS;
+    s32 pad[2];
+    s32 i;
+
+    Actor_SetScale(&this->actor, this->scale);
+    this->actionFunc(this, globalCtx);
+    func_8002D7EC(&this->actor);
+    this->unk_160[0] = this->actor.world.pos;
+
+    for (i = 4; i >= 0; --i) {
+        this->unk_160[i + 1] = this->unk_160[i];
+    }
+
+    if (this->unk_15A != 0) {
+        this->unk_15A--;
+    }
+
+    if (this->unk_15C != 0) {
+        this->unk_15C--;
+    }
+
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 5.0f, 5.0f, 10.0f, 0x1D);
+    if (!(this->scale < 0.6f)) {
+        if (this->actionFunc != func_809B2B48) {
```

Comment:
```suggestion
    if (!(this->scale < 0.6f) && (this->actionFunc != func_809B2B48)) {
```
The negated float compare suggests `if(!((this->scale < 0.6f) || (this->actionFunc == func_809B2B48)))` or even `if((this->scale < 0.6f) || (this->actionFunc == func_809B2B48)) {return;}`, though I know we prefer not to have early returns.

Reply:
I can confirm that all the alternatives you gave matches too.

Should I change it to use the early return (`if((this->scale < 0.6f) || (this->actionFunc == func_809B2B48)) {return;}`) or should I avoid that?